### ISSUE TITLE
Vagrant timed out after replacing private key in VB with three nics

### ISSUE THREAD

**Issue by bgblanch** [mitchellh/vagrant#7665]
### Vagrant version

Vagrant 1.8.4 & VirtualBox 5.0.26
Vagrant 1.8.5 & VirtualBox 5.1.2
### Host operating system

Ubuntu 16.04.1
Windows 10 (via Babun)
### Guest operating system

CentOS 7.1
### Vagrantfile

https://github.com/bgblanch/AutomateOracleRacEnv

``` ruby
rac.vm.box = "boxcutter/centos71"
rac.vm.network "forwarded_port", guest: "1521", host: "1521"
rac.vm.network "private_network", ip: "192.168.10.25", auto_config: false
rac.vm.network "private_network", ip: "172.16.100.11", auto_config: false
rac.vm.network "private_network", ip: "192.168.10.27", auto_config: false

rac.vm.provision "shell", run: "always", inline: "ifconfig enp0s8 192.168.10.25 netmask  255.255.255.0 up"
rac.vm.provision "shell", run: "always", inline: "ifconfig enp0s9 172.16.100.11 netmask  255.255.255.0 up"
rac.vm.provision "shell", run: "always", inline: "ifconfig enp0s10 192.168.10.27 netmask  255.255.255.0 up"
```
### Debug output

https://gist.github.com/bgblanch/517e78b617dfd1c665645b7e8fba61ab

```
==> rac1: Importing base box 'boxcutter/centos71'...
==> rac1: Matching MAC address for NAT networking...
==> rac1: Checking if box 'boxcutter/centos71' is up to date...
==> rac1: Setting the name of the VM: rac1
==> rac1: Clearing any previously set network interfaces...
==> rac1: Preparing network interfaces based on configuration...
    rac1: Adapter 1: nat
    rac1: Adapter 2: hostonly
    rac1: Adapter 3: hostonly
    rac1: Adapter 4: hostonly
==> rac1: Forwarding ports...
    rac1: 1521 (guest) => 1521 (host) (adapter 1)
    rac1: 22 (guest) => 2222 (host) (adapter 1)
==> rac1: Running 'pre-boot' VM customizations...
==> rac1: Booting VM...
==> rac1: Waiting for machine to boot. This may take a few minutes...
    rac1: SSH address: 127.0.0.1:2222
    rac1: SSH username: vagrant
    rac1: SSH auth method: private key
    rac1: Warning: Connection timeout. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```
### Expected behavior

NICs are created and vagrant is able to full provision the VM
### Actual behavior

NICs are created but Vagrant is unable to log into the VM after updating the private key.

I tried the `auto_config: false` but am getting stuck after the insecure key replacement.  Vagrant isn't able to reconnect.  If I comment out the third nic it works but I need the third nic for my project.  I've included the relevant portions.

I've also tried changing the config.vm.boot timeout value to 10 and 20 minutes but it is still timing out.
### Steps to reproduce
1. vagrant up

**Comment by dkarlovi** [comment]
Probably related to #7667.

**Comment by bgblanch** [comment]
![vagrant_authorized_keys](https://cloud.githubusercontent.com/assets/9014841/17329059/751903de-5886-11e6-8d46-670b70864f97.png)
This was taken using Vagrant 1.8.4 and VirtualBox 5.0.26

**Comment by AlphaCactus** [comment]
Related to #7673 ?

**Comment by bgblanch** [comment]
Doesn't appear to be related to 7673.  That one has to do with authorized keys but I believe this is a networking bug.  It works fine when I'm using two vNICs but not three.

**Comment by chrisroberts** [comment]
@bgblanch Can you confirm if this is still an issue within the latest vagrant release? Thanks!

**Comment by bgblanch** [comment]
I'm having some networking conflicts with VMware Workstation.  I'll try this on my home box this weekend.

**Comment by bgblanch** [comment]
Still no luck with Vagrant 1.9.1 and Virtualbox 5.1.12

==> rac1: Importing base box 'bento/centos-7.2'...
==> rac1: Matching MAC address for NAT networking...
==> rac1: Checking if box 'bento/centos-7.2' is up to date...
==> rac1: Setting the name of the VM: rac1
==> rac1: Clearing any previously set network interfaces...
==> rac1: Preparing network interfaces based on configuration...
    rac1: Adapter 1: nat
    rac1: Adapter 2: hostonly
    rac1: Adapter 3: hostonly
    rac1: Adapter 4: hostonly
==> rac1: Forwarding ports...
    rac1: 1521 (guest) => 1521 (host) (adapter 1)
    rac1: 22 (guest) => 2222 (host) (adapter 1)
==> rac1: Running 'pre-boot' VM customizations...
==> rac1: Booting VM...
==> rac1: Waiting for machine to boot. This may take a few minutes...
    rac1: SSH address: 127.0.0.1:2222
    rac1: SSH username: vagrant
    rac1: SSH auth method: private key
    rac1: Warning: Remote connection disconnect. Retrying...
    rac1: Warning: Remote connection disconnect. Retrying...
    rac1: Warning: Remote connection disconnect. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.

**Comment by chrisroberts** [comment]
Hi there,

It looks like this has been resolved within a previously shipped version of Vagrant so I am now closing this issue. If the original issue was not fully resolved, please reopen this issue or create a new one.

Cheers!

**Comment by bgblanch** [comment]
Please reopen this issue.  The last test was with Vagrant 1.9.1 and it wasn't working (see post from 1/10/2017).

**Comment by chrisroberts** [comment]
Hi there. This should be fixed in the latest Vagrant release (1.9.2) which included a fix for private_network configurations on rhel guests.
